mimic/rest/maas_api.py

```diff
@@ -0,0 +1,960 @@
+                del c['entity_id']
+                checks.append(c)
+        metadata = {}
+        metadata['count'] = len(checks)
+        metadata['limit'] = 1000
+        metadata['marker'] = None
+        metadata['next_marker'] = None
+        metadata['next_href'] = None
+        request.setResponseCode(200)
+        return json.dumps({'metadata': metadata, 'values': checks})
+
+    @app.route('/v1.0/<string:tenant_id>/entities/<string:entity_id>', methods=['PUT'])
+    def update_entity(self, request, tenant_id, entity_id):
+        """
+        Update entity. I really just delete it and then put a new one, but with the same id
+        so I don't mess up relationships to checks & alarms
+        """
+        newentity = createEntity(json.loads(request.content.read()))
+        newentity['id'] = entity_id
+        for k in newentity.keys():
+            if 'encode' in dir(newentity[k]):  # because there are integers sometimes.
+                newentity[k] = newentity[k].encode('ascii')
+        for q in range(len(self._entity_cache_for_tenant(tenant_id).entities_list)):
+            if self._entity_cache_for_tenant(tenant_id).entities_list[q]['id'] == entity_id:
+                del self._entity_cache_for_tenant(tenant_id).entities_list[q]
+                self._entity_cache_for_tenant(tenant_id).entities_list.append(newentity)
+                break
+        myhostname_and_port = 'http://' + request.getRequestHostname() + ':' + self.endpoint_port
+        request.setResponseCode(204)
+        request.setHeader('location', myhostname_and_port + request.path + '/' + newentity['id'])
+        request.setHeader('x-object-id', newentity['id'])
+        request.setHeader('content-type', 'text/plain')
+        return ''
+
+    @app.route('/v1.0/<string:tenant_id>/entities/<string:entity_id>', methods=['DELETE'])
+    def delete_entity(self, request, tenant_id, entity_id):
+        """
+        Delete an entity, all checks that belong to entity, all alarms that belong to those checks
+        """
+        entities = self._entity_cache_for_tenant(tenant_id).entities_list
+        checks = self._entity_cache_for_tenant(tenant_id).checks_list
+        alarms = self._entity_cache_for_tenant(tenant_id).alarms_list
+        for e in range(len(entities)):
+            if entities[e]['id'] == entity_id:
+                del entities[e]
+                break
+        for c in checks:
+            if c['entity_id'] == entity_id:
+                del checks[checks.index(c)]
+        for a in alarms:
+            if a['entity_id'] == entity_id:
+                del alarms[alarms.index(a)]
+        request.setResponseCode(204)
+        request.setHeader('content-type', 'text/plain')
+
+    @app.route('/v1.0/<string:tenant_id>/entities/<string:entity_id>/checks', methods=['POST'])
+    def create_check(self, request, tenant_id, entity_id):
+        """
+        Create a check
```

Comment:
any docstring which is just the name of the method with an article inserted probably needs a little work :).  Perhaps a link to the docs, or a description of the `:param:`s, or an explanation of what a "check" represents, or how mimic stores it... something that gives an explanation of the method for future maintainers.